text/0000-third-party-web-components-support.md

```diff
@@ -0,0 +1,103 @@
+property, we would need to support both mechanisms. There are multiple approaches we could take
+here, and the following options are how some popular frameworks do it today. We could default to
+either attributes or properties and introduce a directive that would act as a signal to the engine
+to use one or the other, we could implement a runtime heuristic to choose one over the other, or we
+could default to attributes and set properties when setting a data structure.
+
+#### Proposed solution
+
+When passing data to a native web component, we can take a heuristical approach where if the
+property exists, the property is set. Otherwise the attribute is set. While it may be advantageous
+to require an explicit signal in the form of a directive, this approach seems to strike a good
+balance between developer experience and API semantics.
+
+### Events
+
+EventTarget polyfilled methods and properties such as `add/removeEventListener` and `target` go
+through the synthetic shadow polyfill due to those methods and properties being globally patched.
+This is the current status quo for native shadow roots and should not be an issue for supporting
+third party web components.
+
+LWC's declarative event bindings only support lowercase events. Events using anything other than
+lowercase must be listened to imperatively. This would not need to change in order to support native
+web components.
+
+### Serverside rendering (SSR)
+
+LWC's SSR implementation is tailored specifically for LWC and may not be compatible with a native
+web component implementation. We would probably need to implement the entire synchronous DOM API if
+we wanted to support SSR for native web components.
+
+## Drawbacks
+
+As with any new feature, adding support for native web components introduces additional complexity.
+
+From the platform perspective, there are a few things that Locker vNext needs to implement in order
+to GA this feature. 1) Lift the restriction on `customElements.define()` and 2) provide the
+virtualization mechanism to allow different namespaces to use the same component name.
+
+## Alternatives
+
+No alternatives were considered for the constructor resolution portion of this proposal. An initial
+uninformed idea to introduce error handling for implicit constructor imports was quickly eliminated
+due to the fact that dynamic imports are promise-based.
+
+Alternatives for passing data have already been discussed.
+
+## Adoption strategy
+
+Not much would be required to "sell" this feature as it is something that developers have been
+asking for. Documentation of the new `lwc:external` directive would be required.
+
+# Unresolved questions
+
+- More of a platform-related question, but do we need to make metadata adjustments to exclude native
+  web components from referential integrity?
+- Investigate whether the synthetic shadow slotting implementation is compatible with native web

```

Comment:
I beleive we can assume that because an external custom element is NOT supposed to use synthetic shadow (which I think we can probably enforce somehow... or at least error out), we should be fine here.